src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/VectorOfTFieldLayoutAlgorithm.cs

```diff
@@ -36,6 +36,10 @@ public override ComputedInstanceFieldLayout ComputeInstanceLayout(DefType defTyp
             {
                 instanceFieldSize = new LayoutInt(32);
             }
+            else if (targetDetails.MaximumSimdVectorLength == SimdVectorLength.Vector512Bit)
```

Comment:
Please don't make changes like this without testing that these changes have the expected impact on field layout.

Reply:
Added tests mirroring the existing Vector128<T> and int8x16x2 tests.